api/v1/db/json/pets-dao-example.js

```diff
@@ -46,4 +46,57 @@ const getPetById = id => new Promise((resolve, reject) => {
   }
 });
 
-module.exports = { getPets, getPetById };
+/**
+ * Posts a new pet
+ *
+ * @param {object} body Request body
+ * @returns {Promise} Promise object represents ?
+ */
+const postPet = body => new Promise((resolve, reject) => {
+  try {
+    const rawPets = readJsonFile(dbPath).pets;
+    const { name, owner, species } = body.data.attributes;
+
+    // Determine smallest unused id
+    const ids = _.mapValues(rawPets, 'id');
+    const idHash = {};
+    _.forEach(ids, (id) => {
+      idHash[id] = true;
+    });
+    let id = 1;
+    while (idHash[id]) {
+      id += 1;
+    }
+    rawPets.push({
+      id, name, owner, species,
+    });
+    writeJsonFile(dbPath, { pets: rawPets });
+    resolve(getPetById(id));
+  } catch (err) {
+    reject(err);
```

Comment:
I'm pretty sure that in the promise constructor function argument, if a promise is rejected or an error is thrown, the result will be a rejected promise. So `try { ... } catch (err) { reject(err); }` shouldn't be necessary. The same goes for `try { ... } catch (err) { throw err; }` with `async` functions. Could you also fix this in the other dao methods?